src/ripemd160.rs

```diff
@@ -23,28 +23,30 @@ use std::io;
 
 use byteorder::{ByteOrder, LittleEndian};
 
-use {Error, Hash, HashEngine};
+use Engine as EngineTrait;
+use Hash as HashTrait;
+use Error;
 
 const BLOCK_SIZE: usize = 64;
 
 /// Engine to compute RIPEMD160 hash function
-pub struct Ripemd160Engine {
+pub struct Engine {
```

Comment:
I'd rather rename it to `HashEngine`. I'm sure someone will just import `some_hash::*` and then I'll be annoyed to not really know what an `Engine` is. It's much better to know "this is an engine for the only hash function used in this module" than "this is an engine (steam, electrical, hash, idk)".